analysis/analyze_results_directory.py

```diff
@@ -1,81 +1,88 @@
 import glob
 import dill as pkl
 import numpy as np
 import pandas as pd
 from predicators.src.settings import CFG
 
+GROUPS = [
+    # "ENV",
+    # "APPROACH",
+    # "EXCLUDED_PREDICATES",
+    "EXPERIMENT_ID",
+    # "NUM_TRAIN_TASKS",
+    # "CYCLE"
+]
 
-def create_dataframes() -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
-    """Returns means, standard deviations, and sizes."""
+COLUMN_NAMES_AND_KEYS = [
+    # ("ENV", "env"),
+    # ("APPROACH", "approach"),
+    # ("EXCLUDED_PREDICATES", "excluded_predicates"),
+    ("EXPERIMENT_ID", "experiment_id"),
+    ("SEED", "seed"),
+    # ("NUM_TRAIN_TASKS", "num_train_tasks"),
+    # ("CYCLE", "cycle"),
+    ("NUM_SOLVED", "num_solved"),
+    ("AVG_NUM_PREDS", "avg_num_preds"),
+    ("AVG_TEST_TIME", "avg_suc_time"),
+    ("AVG_SKELETONS", "avg_num_skeletons_optimized"),
+    ("MIN_SKELETONS", "min_skeletons_optimized"),
+    ("MAX_SKELETONS", "max_skeletons_optimized"),
+    ("AVG_NODES_EXPANDED", "avg_num_nodes_expanded"),
+    ("AVG_NODES_CREATED", "avg_num_nodes_created"),
+    ("AVG_NUM_NSRTS", "avg_num_nsrts"),
+    ("AVG_DISCOVERED_FAILURES", "avg_num_failures_discovered"),
+    ("AVG_PLAN_LEN", "avg_plan_length"),
+    ("AVG_EXECUTION_FAILURES", "avg_execution_failures"),
+    ("NUM_TRANSITIONS", "num_transitions"),
+    ("LEARNING_TIME", "learning_time"),
+]
+
+
+def create_dataframes(
+        column_names_and_keys: Sequence[Tuple[str, str]], groups: Sequence[str]
+) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    """Returns means, standard deviations, and sizes.
+
+    When included, config_keys_to_include is a dict from column display
+    name to config name (in CFG).
+    """
     all_data = []
-    column_names = [
-        "ENV", "APPROACH", "EXCLUDED_PREDICATES", "EXPERIMENT_ID", "SEED",
-        "CYCLE", "NUM_SOLVED", "AVG_TEST_TIME", "AVG_SKELETONS",
-        "MIN_SKELETONS", "MAX_SKELETONS", "AVG_NODES_EXPANDED",
-        "AVG_NODES_CREATED", "AVG_NUM_NSRTS", "AVG_DISCOVERED_FAILURES",
-        "AVG_PLAN_LEN", "AVG_EXECUTION_FAILURES", "AVG_NUM_PREDS",
-        "LEARNING_TIME", "NUM_TRANSITIONS"
-    ]
-    groups = [
-        "ENV", "APPROACH", "EXCLUDED_PREDICATES", "EXPERIMENT_ID", "CYCLE"
-    ]
-    some_nonempty_experiment_id = False
+    column_names = [c for (c, _) in column_names_and_keys]
+    for group in groups:
+        assert group in column_names, f"Missing column {group}"
     for filepath in sorted(glob.glob(f"{CFG.results_dir}/*")):
         with open(filepath, "rb") as f:
-            run_data_defaultdict = pkl.load(f)
+            outdata = pkl.load(f)
+        if "config" in outdata:
+            config = outdata["config"].__dict__.copy()
+            run_data_defaultdict = outdata["results"]
+            assert not set(config.keys()) & set(run_data_defaultdict.keys())
+            run_data_defaultdict.update(config)
+        else:
+            print("WARNING: results are old format. Support ending soon.")
```

Comment:
are we really planning to end support for this? if not, let's not say that?

Reply:
for example we want to be able to download old results files and still run this script